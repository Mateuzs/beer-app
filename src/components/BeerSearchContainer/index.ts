export { default as BeerSearchContainer } from "./BeerSearchContainer";
