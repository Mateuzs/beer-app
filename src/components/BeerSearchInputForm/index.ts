export { default as BeerSearchInputForm } from "./BeerSearchInputForm";
