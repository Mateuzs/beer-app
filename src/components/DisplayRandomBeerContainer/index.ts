export { default as DisplayRandomBeerContainer } from "./DisplayRandomBeerContainer";
