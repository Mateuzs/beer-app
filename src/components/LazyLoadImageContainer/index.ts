export { default as LazyLoadImageContainer } from "./LazyLoadImageContainer";
