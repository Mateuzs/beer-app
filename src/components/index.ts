export { BeerSearchContainer } from "./BeerSearchContainer";
export { DisplayRandomBeerContainer } from "./DisplayRandomBeerContainer";
export { NavBar } from "./NavBar";
export { SearchButton } from "./SearchButton";
export { LazyLoadImageContainer } from "./LazyLoadImageContainer";
export { ErrorMessage } from "./ErrorMessage";
export { BeerSearchInputForm } from "./BeerSearchInputForm";
export { BeerList } from "./BeerList";
