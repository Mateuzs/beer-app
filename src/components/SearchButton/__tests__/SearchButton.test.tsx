// external
import React from "react";
import { shallow, mount } from "enzyme";
// components
import { SearchButton } from "../../../components";

describe("SearchButton component", () => {
  test("should match the snapshot and styles", () => {
    const mockedData = {
      label: "BUTTON_LABEL",
      onClickHandler: jest.fn(),
      isDisabled: false,
    };
    const wrapper = shallow(<SearchButton {...mockedData} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("should render expected label and trigger handler when enabled", () => {
    const mockedHandler = jest.fn();
    const mockedData = {
      label: "BUTTON_LABEL",
      onClickHandler: mockedHandler,
      isDisabled: false,
    };
    const component = mount(<SearchButton {...mockedData} />);

    expect(component.find('[data-test-id="search-button"]').text()).toEqual("BUTTON_LABEL");
    expect(component.find("button").hasClass("search-button")).toEqual(true);
    component.simulate("click");
    expect(mockedHandler).toHaveBeenCalled();
  });

  test("should render expected label and not handler when disabled", () => {
    const mockedHandler = jest.fn();
    const mockedData = {
      label: "BUTTON_LABEL",
      onClickHandler: mockedHandler,
      isDisabled: true,
    };
    const component = mount(<SearchButton {...mockedData} />);

    expect(component.find('[data-test-id="search-button"]').text()).toEqual("BUTTON_LABEL");
    expect(component.find("button").hasClass("search-button-disabled")).toEqual(true);
    component.simulate("click");
    expect(mockedHandler).toHaveBeenCalledTimes(0);
  });
});
