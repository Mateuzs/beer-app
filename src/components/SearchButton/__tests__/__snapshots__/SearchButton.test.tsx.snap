// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton component should match the snapshot and styles 1`] = `
<button
  className="search-button"
  data-test-id="search-button"
  disabled={false}
  onClick={[MockFunction]}
>
  BUTTON_LABEL
</button>
`;
